dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
-
-            return Response.ok().build();
-        } catch(DotDataException dt){
-        	Logger.error(this, dt.getMessage());
-        	throw new BadRequestException(dt, dt.getMessage());
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to update index replicas: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
+    @Path("/activate/{params:.*}")
+    public Response activateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
+        InitDataObject init=auth(httpServletRequest,httpServletResponse);
+        String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+        return modIndex(httpServletRequest, httpServletResponse, indexName, IndexAction.ACTIVATE.name());
+    }
+    
+    
+    /**
+     * use modIndex instead, e.g. PUT /api/v1/esindex/{indexName}?action=deactivate
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param params
+     * @return
+     */
+    @Deprecated
+    @PUT
+    @Path("/deactivate/{params:.*}")
+    public Response deactivateIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
+        InitDataObject init=auth(httpServletRequest,httpServletResponse);
+        String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
+        return modIndex(httpServletRequest, httpServletResponse, indexName, IndexAction.DEACTIVATE.name());
     }
 
+    /**
+     * use modIndex instead, e.g. PUT /api/v1/esindex/{indexName}?action=close
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param params
+     * @return
+     */
+    @Deprecated
     @PUT
     @Path("/close/{params:.*}")
-    public Response closeIndex(@Context HttpServletRequest httpServletRequest,@Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
-        try {
-            InitDataObject init=auth(params,httpServletRequest,httpServletResponse);
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
-            APILocator.getESIndexAPI().closeIndex(indexName);
-
-            return Response.ok().build();
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to open index: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
+    public Response closeIndex(@Context HttpServletRequest httpServletRequest,@Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'httpServletRequest' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)